mobile/lib/utils/http_ssl_cert_override.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:io';
+import 'package:immich_mobile/modules/settings/services/app_settings.service.dart';
+import 'package:immich_mobile/shared/models/store.dart';
+import 'package:logging/logging.dart';
+
+class HttpSSLCertOverride extends HttpOverrides {
+  @override
+  HttpClient createHttpClient(SecurityContext? context) {
+    return super.createHttpClient(context)
+      ..badCertificateCallback = (X509Certificate cert, String host, int port) {
+        var log = Logger("HttpSSLCertOverride");
+
+        AppSettingsEnum setting = AppSettingsEnum.allowSelfSignedSSLCert;
+        
+        // Check if user has allowed self signed SSL certificates.
+        bool selfSignedCertsAllowed =
+            Store.get(setting.storeKey as StoreKey<bool>, setting.defaultValue);
+
+        bool isLoggedIn = Store.tryGet(StoreKey.currentUser) != null;
+
+        // Conduct server host checks if user is logged in to avoid making
+        // insecure SSL connections to services that are not the immich server.
+        if (isLoggedIn && selfSignedCertsAllowed) {
```

Comment:
It looks like this only bypasses after the user is logged in. What about when the user is logging in for the first time? Is it not possible to use a self signed certificate then?

Reply:
If the user is not logged then it will allow the use of the self-signed cert based on if the user has enabled this functionality.

The if statement is only to do hostname checks when the user is logged in